python/server/ServerWrapper.cpp

```diff
@@ -475,18 +524,8 @@ namespace {
         int SpawnLimit()
         { return m_monster_fleet_plan->SpawnLimit(); }
 
-        bool Location(int object_id) {
-            // get the universe object to test and check if it exists
-            TemporaryPtr<UniverseObject> obj = GetUniverseObject(object_id);
-            if (!obj) {
-                ErrorLogger() << "MonsterFleetPlanWrapper::Location: Couldn't get object with ID " << object_id;
-                return false;
-            }
-
-            // get location condition and evaluate it with the specified universe object
-            // if no location condition has been defined, any object matches
-            const Condition::ConditionBase* location_test = m_monster_fleet_plan->Location();
-            return (!location_test || location_test->Eval(obj));
+        list Locations(list systems) {
+            return FilterIDsWithCondition(m_monster_fleet_plan->Location(), systems);
```

Comment:
Why *replace* `Location` with `Locations`? I'd keep the former and just add the latter, both are useful.

Having both functions (`Location` to test a single object against the location condition and `Locations` to get a list of all objects that match a location condition) for the specials would be a good idea too (if you're willing to add that one too).